atomic_reactor/plugins/pre_pyrpkg_fetch_artefacts.py

```diff
@@ -26,13 +29,59 @@ def __init__(self, workflow):
         constructor
 
         :param workflow: DockerBuildWorkflow instance
-        :param command: str, command to use to get artefacts (e.g. 'make sources')
-                             it is executed in cloned git repo
+        :type workflow: atomic_reactor.inner.DockerBuildWorkflow
         """
         # call parent constructor
         super(DistgitFetchArtefactsPlugin, self).__init__(workflow)
         self.command = self.workflow.conf.sources_command
 
+    def _collect_exploded_sources_files(self, build_dir_path: Path) -> List[Path]:
+        """Collect the fetched exploded sources.
+
+        :param build_dir_path: from which directory to find the exploded sources file.
+        :type build_dir_path: pathlib.Path
+        """
+        source_repos = build_dir_path / 'source-repos'
+        filenames = []
+        if not source_repos.exists():
+            return filenames
+        with source_repos.open('r', encoding='utf-8') as f:
+            for line in f:
+                git_ref = line.rstrip().split()[-1]
+                filenames.append(next(build_dir_path.glob(f'*-{git_ref}.tar.gz')))
+        return filenames
+
+    def _fetch_sources(self, build_dir: BuildDir) -> List[Path]:
+        """Fetch sources files.
+
+        :param build_dir: download the sources files into this directory.
+        :type build_dir: BuildDir
+        :return: a list of downloaded file names.
+        :rtype: list[pathlib.Path]
+        """
+        # Create a dedicated directory to hold the fetched sources files, from
+        # where to generated the downloaded file list.
+        build_dir_path = build_dir.path
+        sources_outdir = build_dir_path / 'outdir'
+        sources_outdir.mkdir()
+
+        sources_cmd = self.command.split()
+        sources_cmd.append('--outdir')
+        sources_cmd.append(str(sources_outdir))
+        self.log.debug('Fetching sources: %r', sources_cmd)
+        subprocess.check_call(sources_cmd, cwd=build_dir_path)
+
+        fetched_sources = []
+        for file_name in sources_outdir.iterdir():
+            shutil.move(str(file_name), str(build_dir_path))
+            fetched_sources.append(file_name.relative_to(sources_outdir))
+        sources_outdir.rmdir()
+
+        # Collect the exploded sources if there is
+        fetched_sources += self._collect_exploded_sources_files(build_dir_path)
```

Comment:
It feels odd that some files will be fetched into the build_dir itself while others will be fetched into the new empty dir and then moved. I would also prefer not to parse any rpkg-specific files if possible.

Could the following approach work?
```
files_before = set(all files from build_dir)
run rpkg in build_dir, fetch everything into build_dir
files_after = set(all files from build_dir)

fetched_files = files_after - files_before
```
(this has the potential problem with existing files - e.g. if ./some/file exists, is it possible that rpkg will overwrite ./some/file with a new one?)

Reply:
Discussed offline, this is probably not worth doing